### ISSUE TITLE
column names and values mixed up

### ISSUE THREAD

**Issue by jkucan** [noplay/python-mysql-replication#282]
**Note**: this is only observable with MySql 8 (and not with 5.7.25).

When reading `WriteRowsEvent` the `rows` attribute has mixed up column names and values. Specifically when inserting row:
```
INSERT INTO BUG.test_pets (id, name, kind) VALUES (1, 'rex', 'dog')
```

`rows` in the event is
```
{"values": {"id": 1, "kind": "rex", "name": "dog"}}
```
Note the values if `kind` and `name` swapped.
Here the code to reproduce the issue.
```
"""
To see the issue start MySQL server version 8 with

    $ docker run --name mysql -p 3306:3306 -e MYSQL_ROOT_PASSWORD=root -d mysql --log-bin --binlog-format=ROW --gtid-mode=ON --enforce-gtid-consistency=TRUE

"""

import mysql.connector
import logging
import json
import datetime
from pymysqlreplication import BinLogStreamReader
from pymysqlreplication.row_event import WriteRowsEvent

logging.info("start")
log = logging.getLogger(__name__)
log.setLevel('DEBUG')

def ts2str(timestamp):
    """
    Convert unit timestamp to string representation.
    """
    return datetime.datetime.utcfromtimestamp(timestamp).isoformat(sep=' ')

connection = mysql.connector.connect(
    host='localhost',
    user='root',
    port=3306,
    password='root',
    database='sys',
    )

cursor = connection.cursor()

# set up database
cursor.execute("DROP DATABASE IF EXISTS BUG")
cursor.execute("CREATE DATABASE BUG")
cursor.execute("USE BUG")
cursor.execute("CREATE TABLE BUG.test_pets (id INT, name VARCHAR(20), kind VARCHAR(20))")
connection.commit()

# get the last GTID
cursor.execute("SHOW VARIABLES LIKE 'gtid_executed'")
name, gtid = cursor.fetchone()

# open the binlog stream
stream = BinLogStreamReader(
    connection_settings={
        "host": 'localhost',
        "port": 3306,
        "user": 'root',
        "passwd": 'root',
    },
    server_id=100,
    blocking=True,
    resume_stream=True,
    only_events=[WriteRowsEvent],
    auto_position=gtid,
)

# execute an insert
cursor.execute("INSERT INTO BUG.test_pets (id, name, kind) VALUES (1, 'rex', 'dog')")
connection.commit()

# read the binlog
for binlogevent in stream:
    log.debug(f'Received {type(binlogevent).__name__}.')
    for row in binlogevent.rows:
        log.debug(f'{ts2str(binlogevent.timestamp)}: {json.dumps(row)}')
    break

```

**Comment by furuiyang0715** [comment]
Yes, I  have a same problem with MySql 8...

**Comment by muayyad-alsadi** [comment]
please note that there are different binlog  formats and different formats for the raw events:

```
--log-bin-use-v1-row-events[={0|1}]

Version 2 binary log row events are available beginning with MySQL 5.6.6; however, Version 2 events cannot be read by previous MySQL Server releases. Setting this option to 1 causes mysqld to write the binary log using Version 1 logging events, which is the only version of binary log events used in previous releases, and thus produce binary logs that can be read by older slaves. Setting --log-bin-use-v1-row-events to 0 (the default) causes mysqld to use Version 2 binary log events. 
```

* https://dev.mysql.com/doc/refman/5.6/en/replication-options-binary-log.html#option_mysqld_log-bin-use-v1-row-events
* https://dev.mysql.com/doc/internals/en/binary-log-versions.html

**Comment by muayyad-alsadi** [comment]
when we set configuration to

```
log_bin_use_v1_row_events = 1
```

names and values did not mix, but names and types seems to be still mixed.

note: minor versions of mysql 8.0.x seems to be not compatible between each-other.

**Comment by subnix** [comment]
I have the same problem, but it seems to be fixed in #271.
@noplay, can you update the pypi package?

**Comment by noplay** [comment]
0.19 released :D
